### PR TITLE
chore: update to yocto 4.0.17

### PR DESCRIPTION
**Pull request by HarryWaschkeit**
- u-boot/phytec (tauril2): adapted bootloader checksum for  bootloader
- class omnect_bootloader_versioning: for better readability output hash first in checksum file
- kas:
  - updated bitbake to tag ref yocto-4.0.17
  - updated openebedded-core to tag ref yocto-4.0.17
    (https://downloads.yoctoproject.org/releases/yocto/yocto-4.0.17/RELEASENOTES)
  - updated meta-openembedded to latest kirkstone HEAD
  - updated meta-phytec to latest kirkstone HEAD
  - updated meta-freescale to latest kirkstone HEAD
  - updated meta-freescale-distro to latest kirkstone HEAD
  - updated meta-raspberrypi to latest kirkstone HEAD
  - updated meta-swupdate to latest kirkstone HEAD
  - updated meta-virtualization to latest kirkstone HEAD
  - updated meta-yocto-bsp (poky) to latest kirkstone HEAD
  - adapted layer patches as necessary

### CODE REVIEW

**Review by JanZachmann — APPROVED** [review]

File: kas/patches/meta-phytec_layerdir.patch [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-index a14101de..23762ded 100644
+index 29ac1c3e..9ff4cc05 100644
 --- a/conf/layer.conf
 +++ b/conf/layer.conf
-@@ -36,3 +36,5 @@ BBFILES_DYNAMIC += " \
+@@ -46,3 +46,5 @@ LAYERRECOMMENDS_phytec = "\
  "
  
- BBMASK:mx8m-nxp-bsp:append = " meta-imx/meta-bsp/recipes-graphics/imx-gpu-apitrace/imx-gpu-apitrace_10.0.0.bb"
+ BBMASK:append:mx8m-nxp-bsp = " meta-imx/meta-bsp/recipes-graphics/imx-gpu-apitrace/imx-gpu-apitrace_10.0.0.bb"
```
Comment (JoergZeidler) [inline kas/patches/meta-phytec_layerdir.patch:8]: why is that changed?

**Review by JoergZeidler — APPROVED** [review]
